tensorflow/lite/kernels/reduce.cc

```diff
@@ -282,7 +283,54 @@ TfLiteStatus PrepareMeanOrSum(TfLiteContext* context, TfLiteNode* node) {
     return kTfLiteOk;
   }
   temp_sum->allocation_type = kTfLiteArenaRw;
-  return ResizeTempSum(context, &op_context, temp_sum);
+  return ResizeTempAccum(context, &op_context, temp_sum);
+}
+
+float GetQuantProdScaling(float input_scale, float output_scale,
+                          int reduced_axis_size) {
+  // The scaling after taking the product of all the quantized values should
+  // be (input_scale**reduced_axis_size)/output_scale but to avoid overflowing
+  // the accumulator we instead scale each multiplication by
+  // input_scale/nth_root(output_scale, reduced_axis_size).
+  return input_scale / std::pow(output_scale, 1.0 / reduced_axis_size);
```

Comment:
Can we follow other kernels and use double for effective scale?

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/kernels/kernel_util.cc#L254-L256

Reply:
Effectively I was using float as `params.scale` is a float but double should be preferred.

Reply:
Ah... Thanks. I missed the input type.